tests/tensor/test_basic.py

```diff
@@ -4310,3 +4312,98 @@ def test_oriented_stack_functions(func):
+        self.op = convolve
+        self.op_class = Convolve
+
+    def _validate_output(self, a, v, mode):
+        aesara_sol = self.op(a, v, mode=mode).eval()
+        numpy_sol = np.convolve(a, v, mode=mode)
+        assert np.allclose(numpy_sol, aesara_sol)
+
+    @pytest.mark.parametrize(
+        "a, v, mode",
+        [
+            (
+                np.arange(3).astype(config.floatX),
+                np.arange(6).astype(config.floatX),
+                "full",
+            ),
+            (
+                np.arange(5).astype(config.floatX),
+                np.arange(3).astype(config.floatX),
+                "valid",
+            ),
+            (
+                np.arange(8).astype(config.floatX),
+                np.arange(9).astype(config.floatX),
+                "same",
+            ),
+            (np.arange(3).astype(int), np.arange(6).astype(int), "full"),
+            (
+                np.random.normal(size=4).astype(config.floatX),
+                np.random.uniform(size=8).astype(int),
+                "valid",
+            ),
+        ],
+    )
+    def test_op(self, a, v, mode):
+        self._validate_output(a, v, mode)
+
+    def test_scalar_error(self):
+        with pytest.raises(ValueError, match="must be 1-dim"):
+            self.op(4, [4, 1])
+
+    @pytest.mark.parametrize(
+        "a_shape, v_shape, mode, exp_res, error_regex",
+        [
+            ((1,), (3,), "full", (3,), None),
+            ((2,), (3, 1), "full", None, "must be 1-dim"),
+            ((2,), (3,), "valid", (2,), None),
+            ((2,), (3,), "same", (3,), None),
+            ((2,), (3,), "val", None, "must be full, valid or same"),
+        ],
+    )
+    def test_get_output_shape(self, a_shape, v_shape, mode, exp_res, error_regex):
+        a = tensor(dtype=np.float64, shape=a_shape)
+        v = tensor(dtype=np.float64, shape=v_shape)
+
+        if error_regex is not None:
+            with pytest.raises(ValueError, match=error_regex):
+                self.op_class._get_output_shape(
+                    a, v, (a_shape, v_shape), mode, validate=True
+                )
+            assert (
+                self.op_class._get_output_shape(
+                    a, v, (a_shape, v_shape), mode, validate=False
+                )
+                == ()
+            )
+        else:
+            assert (
+                self.op_class._get_output_shape(
+                    a, v, (a_shape, v_shape), mode, validate=True
+                )
+                == exp_res
+            )
+
+    def test_get_output_shape_symbolic(self):
+        m, n = aesara.tensor.iscalar("m"), aesara.tensor.iscalar("n")
+        a_shape, v_shape = (m,), (n,)
+        a = tensor(dtype=np.float64, shape=a_shape)
+        v = tensor(dtype=np.float64, shape=v_shape)
+        result = self.op_class._get_output_shape(
+            a, v, (a_shape, v_shape), "full", validate=True
+        )
+        assert aesara.tensor.eq(result, (m + n - 1))
```

Comment:
These are returning Aesara graphs and evaluating to `True` only because they're non-`None`/"empty".  Instead, let's change these to explicit static `Type`-level shape checks (e.g. `z = convolve(a, v)` and `assert z.type.shape == ...`).  Also, we need to check at least one case in which the static shape isn't constant (i.e. when a static shape value should be `None`).

N.B. You can extend `utt.InferShapeTester` and use its `self._compile_and_check` method to test shape inference (i.e. the `Convolve.infer_shape` implementation).

Reply:
> N.B. You can extend utt.InferShapeTester and use its self._compile_and_check method to test shape inference (i.e. the Convolve.infer_shape implementation).

Done, thanks.

> Instead, let's change these to explicit static Type-level shape checks (e.g. z = convolve(a, v) and assert z.type.shape == ...). 

Is there any way to test two symbolic expressions, as was required in the previous commit? 

> (i.e. when a static shape value should be None).

Can you please provide an example of this? I wasn't able to catch that.